src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -154,3 +196,64 @@ std::optional<Point> CreaseDefenderFSM::findDefenseAreaIntersection(
 
     return std::nullopt;
 }
+
+bool CreaseDefenderFSM::ballNearbyWithoutThreat(const Update& event)
+{
+    Point robot_position = event.common.robot.position();
+    Point ball_position  = event.common.world_ptr->ball().position();
+
+    std::optional<Robot> nearest_friendly =
+        event.common.world_ptr->friendlyTeam().getNearestRobot(ball_position);
+    std::optional<Robot> nearest_enemy =
+        event.common.world_ptr->enemyTeam().getNearestRobot(robot_position);
+
+    if (event.control_params.ball_steal_mode == TbotsProto::BallStealMode::IGNORE)
```

Comment:
New param with modes IGNORE or STEAL to allow us to toggle stealing depending on if it is allowed (eg. we do not want the edge case where a passing by crease defender steals an enemy free kick)